### PR TITLE
enhanced German translation, partly also english

### PR DESCRIPTION
**Pull request by malenki**

### CODE REVIEW

File: data/categories.txt [hunk 1]
```diff
@@ -785,7 +785,7 @@ shop-car
 en:car shop|U+1F697|U+1F698|U+1F699
 ru:автосалон|магазин
 uk:автосалон|магазин
-de:Auto-Shop|Laden
+de:Autohaus
```
Comment (deathbaba) [inline data/categories.txt]: You need to leave |Laden here to match category search (when selecting shop icon in the search).

File: data/categories.txt [hunk 2]
```diff
@@ -1191,7 +1191,7 @@ tourism-information
 en:4tourist information|4information|sights|U+2139|U+1F481
 ru:4туринформация|информация|достопримечательность
 uk:4турінформація|інформація|пам’ятка|пам’ятні місця
-de:4Information|Sehenswürdigkeit
+de:4Information|Tourist-Information
```
Comment (deathbaba) [inline data/categories.txt]: You can swap these words, so users will se first one on a screen (and second and others will be there as search synonyms).

Comment (deathbaba) [inline data/categories.txt]: Word Sehenswürdigkeit also should be there to match category search

File: data/categories.txt [hunk 3]
```diff
@@ -1220,7 +1220,7 @@ tourism-picnic_site|amenity-bbq
 en:3picnic site|sights
 ru:3пикник|достопримечательность
 uk:3пікнік|зупинка|відпочинок|пам’ятні місця
-de:3Picknick|Sehenswürdigkeit
+de:3Picknick|Grillplatz
```
Comment (deathbaba) [inline data/categories.txt]: Please leave |Sehenswürdigkeit at the end for category search.

File: data/categories.txt [hunk 4]
```diff
@@ -1249,7 +1249,7 @@ amenity-place_of_worship
 en:place of worship|4temple|sights|U+1F64F|U+26EA
 ru:3храм|церковь|достопримечательность
 uk:3храм|церква|собор|пам’ятка|алтарь|мавзолей|пам’ятні місця
-de:Anbetungsstätte|Sehenswürdigkeit
+de:Anbetungsstätte|Kirche|Dom|Tempel|Kathedrale|Kapelle|Moschee|Sehenswürdigkeit
```
Comment (deathbaba) [inline data/categories.txt]: Only 'generic' tempe name and synonyms should be here, which fits any religion.
All specific religious temples are at the end of this file.

File: data/categories.txt [hunk 5]
```diff
@@ -1865,7 +1865,7 @@ amenity-fountain
 en:fountain|U+26F2
 ru:фонтан
 uk:фонтан
-de:Brunnen|Fontäne
+de:Springbrunnen|Fontäne
```
Comment (deathbaba) [inline data/categories.txt]: Is it ok to leave Brunnen as a second synonym here?

File: data/categories.txt [hunk 6]
```diff
@@ -2850,7 +2850,7 @@ highway-primary|highway-primary_link|highway-residential|highway-secondary|highw
 en:street|st|str|ave|road|rd
 ru:улица|переулок|проезд|тракт|проспект|ул
 uk:вулиця|провулок|проїзд|проспект|бульвар|вул
-de:Strasse|Weg|Avenue|Gasse|Str
+de:Straße|Weg|Boulevard|Gasse
```
Comment (deathbaba) [inline data/categories.txt]: Str was recently added by another german user. Are you sure it needs to be removed?

**Comment by deathbaba** [comment]
Please also accept our CLA (it's just a formality), so we can merge your PR: http://maps.me/cla/

File: data/categories.txt (hunk 1 above)
Comment (malenki) [inline data/categories.txt]: Ok. There should be some documentation somewhere about how this file is processed. My lack of knowledge also resulted in removing some of the "Sehenswürdigkeit" below.   
Hint: "Geschäft" would IMHO be a better category name then "Laden"

Comment (deathbaba) [inline data/categories.txt]: Is "Geschäft" better as a generic "shop" or "shops" than "Laden"? What is the difference?

File: data/categories.txt (hunk 4 above)
Comment (malenki) [inline data/categories.txt]: Ok, but Chinese and Japanese seem to be quite verbose, too. (I don't speak those languages, though.)

File: data/categories.txt (hunk 5 above)
Comment (malenki) [inline data/categories.txt]: A "Brunnen" or well is the opposite of a fountain. And wouldn't searching for "Brunnen" also match on "Spring**brunnen**?

File: data/categories.txt (hunk 6 above)
Comment (malenki) [inline data/categories.txt]: I thought that searching for Str would match on **Str**aße as well. Is that so?  
Does the program translate "ß" to "ss" and similar stuff? Else I would have to add back "Strasse" … which is just the way of writing "Straße" when people have no "ß" at hand (that is nearly everybody outside Austria/Germany).

File: data/categories.txt (hunk 5 above)
Comment (deathbaba) [inline data/categories.txt]: No, our search matches prefixes only.

File: data/categories.txt (hunk 1 above)
Comment (malenki) [inline data/categories.txt]: Alex Zolotarev schrieb
> 
> Is "Geschäft" better as a generic "shop" or "shops" than "Laden"?
> What is the difference?

"Laden" is usually used for smaller retail businesses, rarely (if all)
for supermarkets/discounters and I'd doubt that people in Germany use it
at all for car dealers.
To describe "Geschäft" quite vaguely: A place where you get $stuff (be
it a cleaning for your tuxedo, a car or a new axe) for money

File: data/categories.txt (hunk 4 above)
Comment (deathbaba) [inline data/categories.txt]: I’ve just cleaned them up - they had many old translations not related to churches.

File: data/categories.txt (hunk 1 above)
Comment (deathbaba) [inline data/categories.txt]: Here is a fresh documentation: https://github.com/mapsme/omim/wiki/categories.txt <https://github.com/mapsme/omim/wiki/categories.txt>

File: data/categories.txt (hunk 6 above)
Comment (deathbaba) [inline data/categories.txt]: Str should match it anyway, I suppose it is also hard-coded somewhere else in the search engine.
Our search should correctly handle ß (you can probably test it by placing categories.txt into app data folder on iTunes or MapsWithMe folder on android).

File: data/categories.txt (hunk 1 above)
Comment (deathbaba) [inline data/categories.txt]: Thanks. we’ll discuss it with our team. For me, small retail businesses are also ok :)

**Comment by deathbaba** [comment]
Can you please rebase your request and fixup (squash) all commits into one? We have just merged missing temples translations.

**Comment by deathbaba** [comment]
Merged in #1767

**Pull request closed without merging**